internal/genny/build/build.go

```diff
@@ -49,7 +49,9 @@ func New(opts *Options) (*genny.Generator, error) {
 	g.Transformer(genny.Replace("/dot-", "/."))
 
 	// validate templates
-	g.RunFn(ValidateTemplates(os.DirFS(opts.App.Root), opts.TemplateValidators))
+	if _, err := os.Stat(opts.App.Root + "/templates"); err == nil {
```

Comment:
don't use string concatination for filepaths. Use `filepath.Join()` instead

Reply:
I think you worry about Windows. That could be generally safe for some situations, but still, golang just supports '/' as a path separator for Windows OS. [1] Also `os` package itself uses this way in their code [2].

Also, TIL, Windows native API also understand '/' as a path separator (and I was told it was true from MS-DOS era). The restriction of '\' is just a restriction of `command.com` :-)

[1] https://github.com/golang/go/blob/a59ab29bf2164730e3e6d5439c6ec4b1e46adcc4/src/os/path_windows.go#L15
[2] https://github.com/golang/go/blob/a59ab29bf2164730e3e6d5439c6ec4b1e46adcc4/src/os/file.go#L658

Reply:
It's not just windows. String concatenation for file paths is generally unsafe. You wouldn't do the same thing with SQL statements would you? 😉 

filepath.Join() sanitizes the input as well (see [here](https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/path/filepath/path_unix.go;l=42;)) and it also works with every OS that golang supports, not just Windows, Linux and Mac.

Furthermore using + for string concatenation is also slow and memory inefficient, because strings are immutable in golang. Here this doesn't matter, but its generally good practice to avoid '+' with strings. I'm actually surprised that [os.Stat](https://github.com/golang/go/blob/a59ab29bf2164730e3e6d5439c6ec4b1e46adcc4/src/os/file.go#L658) uses concatination.

Reply:
Will update the patch as you wish. but the reason still we have `+` for string is that it is still useful :-) For sanitizing or any other additional safety logic, they are needed sometimes especially when we handle unexpected input or output but when it is crystal clear, we can just go straight with simpler code.

Reply:
Another problem is also opt.App.Root may end in a trailing `/` or it might not. On windows it might use `\\` `filepath.Join` handles that case.